pkg/kubelet/kubelet.go

```diff
@@ -2262,6 +2262,7 @@ func (kl *Kubelet) rejectPod(pod *v1.Pod, reason, message string) {
 		Phase:   v1.PodFailed,
 		Reason:  reason,
 		Message: "Pod was rejected: " + message})
+	kl.statusManager.TerminatePod(pod)
```

Comment:
I **think** this should be safe, but it would be great some feedback. I'm thinking of two cases:

(1) The kubelet has never seen this pod ever (in any prior restart) and since the kubelet rejected the pod, it has never been started. This is the case in https://github.com/kubernetes/kubernetes/issues/118472

(2) The more involved case if the pod was started by kubelet in the past, kubelet then restarted, and after restart the kubelet rejected the pod. In this case, I believe `HandlePodCleanups` should terminate the pod as an orphaned runtime pod and in parallel the pod could be marked Terminated here.

Reply:
cc @smarterclayton @mimowo

Reply:
In general it looks like an alternative solution, two thougts:
1. what are the consequences of the situation described in (2.), can we reproduce it to see?
2. can we avoid stepping into (2.) by doing 	
```golang
kl.podWorkers.UpdatePod(UpdatePodOptions{
	Pod:        pod,
	UpdateType: kubetypes.SyncPodKill,
})
```
IIUC it means that we go through `syncTerminatingPod`, but this does not seem like much overhead, and it avoids the pod to be considered orphan, which avoids (2.) and makes reasoning simpler. I'm testing it here: https://github.com/kubernetes/kubernetes/pull/118614

Reply:
@bobbypage I have prepared a test for the scenario when a previously running pod is rejected after kubelet restart. It appears to pass both on my branch and your (but seems for slightly different reasons): https://github.com/kubernetes/kubernetes/pull/118614/commits/d9111b2f000c5189887bc44cd5d666992efcded8